commands/remove.rb

```diff
@@ -34,16 +34,30 @@ def self.remove(pkg, verbose)
     # Remove the files and directories installed by the package.
     unless pkg.is_fake?
       Dir.chdir CREW_CONFIG_PATH do
-        # Remove all files installed by the package.
+        # Remove all files installed by the package in CREW_PREFIX and
+        # HOME unless the file exists in another installed package, or
+        # if the file is part of CREW_ESSENTIAL_FILES, as that has files
+        # not covered by CREW_ESSENTIAL_PACKAGES.
         flist = File.join(CREW_META_PATH, "#{pkg.name}.filelist")
         if File.file?(flist)
-          File.foreach(flist, chomp: true) do |line|
-            next unless line.start_with?(CREW_PREFIX)
-            if system("grep --exclude #{pkg.name}.filelist -Fxq '#{line}' ./meta/*.filelist")
-              puts "#{line} is in another package. It will not be removed during the removal of #{pkg.name}.".orange
+          package_files = []
+          package_files << `grep -h "^#{CREW_PREFIX}\\|^#{HOME}" #{flist}`.split("\n")
+          all_other_files = []
+          all_other_files << `grep -h --exclude #{pkg.name}.filelist "^#{CREW_PREFIX}\\|^#{HOME}" #{CREW_META_PATH}/*.filelist`.split("\n")
+
+          unique_to_package_files = package_files.flatten - all_other_files.flatten
+          package_files_that_overlap = all_other_files.flatten & package_files.flatten
+
+          unless package_files_that_overlap.flatten.empty?
+            puts "The following file(s) in other packages will not be deleted during the removal of #{pkg.name}:".orange
+            puts package_files_that_overlap.join("\n").orange
+          end
+          unique_to_package_files.flatten.each do |file|
+            if CREW_ESSENTIAL_FILES.include?(File.basename(file))
```

Comment:
Couldn't we check if the package is included in the dependencies of `CREW_ESSENTIAL_PACKAGES`? i.e. using `pkg.get_deps_list`

or, given that the dependency problem only applies to `glibc_*` to my knowledge, we could just add a wildcard check for that

Reply:
I think that would be ideal!

Can you figure out how to do that properly?

I couldn't figure out how to use that to pull the package d'EPS of stuff like glibc_lib.rb from inside remove.rb...

Reply:
> Seriously though, how is one supposed to invoke `pkg.get_deps_list` for other packages from elsewhere in `crew`?

We can do it with one line:
```ruby
Package.load_package('pkg.rb').get_deps_list
```

Reply:
@supechicken Looks like that works if I make sure `pkg_file` includes the path to the file. 

E.g., it works if I change `self.load_package` in `package.rb` to have this before `class_eval(File.read(pkg_file`:
```
    # read and eval package script under 'Package' class, using the
    # newest file available.
    pkg_file = ["#{CREW_LOCAL_REPO_ROOT}/packages/#{pkg_name}.rb", "#{CREW_PACKAGES_PATH}/#{pkg_name}.rb"].max { |a, b| File.mtime(a) <=> File.mtime(b) }

```

Reply:
Ok I got it working.